hedgehog/cabal.project

```diff
@@ -0,0 +1,4 @@
+packages: .
+if impl(ghc >= 9.6)
+  package hedgehog
+    ghc-options: -Wwarn=x-partial
```

Comment:
That’s interesting. Have you kept the error produced without this option?

Reply:
The error is in the CI build for this PR so it seems the attempt at disabling it failed to take effect for some reason.

```
src/Hedgehog/Internal/Property.hs:429:31: error: [GHC-63394] [-Wx-partial, -Werror=x-partial]
Error:     In the use of ‘head’
    (imported from Hedgehog.Internal.Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
429 |     groups = List.map (\l -> (head l, length l)) $ List.group sp
    |                               ^^^^
```

Reply:
I guess it failed because `cabal.project` was added in a place where it's not being picked up.

Reply:
I submitted an amended PR that fixes this: https://github.com/hedgehogqa/haskell-hedgehog/pull/504